From 786146bf57b0ea6b18bab422b2c12ac8ff818a95 Mon Sep 17 00:00:00 2001
From: zeta945 <zeta945@gmail.com>
Date: Sat, 31 Dec 2022 23:45:58 +0800
Subject: [PATCH] Merge branch 'master' into patch_base

---
 Modules/getpath_noop.c | 7 +++++++
 configure.ac           | 4 +++-
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/Modules/getpath_noop.c b/Modules/getpath_noop.c
index c10e41d..5d4863a 100644
--- a/Modules/getpath_noop.c
+++ b/Modules/getpath_noop.c
@@ -8,3 +8,10 @@ _PyConfig_InitPathConfig(PyConfig *config, int compute_path_config)
 {
     return PyStatus_Error("path configuration is unsupported");
 }
+
+PyObject *
+_Py_Get_Getpath_CodeObject(void)
+{
+    Py_RETURN_NONE;
+}
+
diff --git a/configure.ac b/configure.ac
index 99d3147..66384ea 100644
--- a/configure.ac
+++ b/configure.ac
@@ -6954,7 +6954,8 @@ dnl AC_MSG_NOTICE([m4_set_list([_PY_STDLIB_MOD_SET_NA])])
 dnl Default value for Modules/Setup.stdlib build type
 AS_CASE([$host_cpu],
   [wasm32|wasm64], [MODULE_BUILDTYPE=static],
-  [MODULE_BUILDTYPE=${MODULE_BUILDTYPE:-shared}]
+  [aarch64], [MODULE_BUILDTYPE=static],
+  [MODULE_BUILDTYPE=${MODULE_BUILDTYPE:-static}]
 )
 AC_SUBST([MODULE_BUILDTYPE])
 
@@ -6963,6 +6964,7 @@ AC_MSG_CHECKING([for additional Modules/Setup files])
 AS_CASE([$ac_sys_system],
     [Emscripten], [MODULES_SETUP_STDLIB=Modules/Setup.stdlib],
     [WASI], [MODULES_SETUP_STDLIB=Modules/Setup.stdlib],
+    [Darwin], [MODULES_SETUP_STDLIB=Modules/Setup.stdlib],
     [MODULES_SETUP_STDLIB=]
 )
 AC_MSG_RESULT([$MODULES_SETUP_STDLIB])
-- 
2.25.1

